target/scripts/startup/setup.d/security/misc.sh

```diff
@@ -270,15 +270,22 @@ function _setup_spam_to_junk
   if [[ ${MOVE_SPAM_TO_JUNK} -eq 1 ]]
   then
     _log 'debug' 'Spam emails will be moved to the Junk folder'
+    cat >/usr/lib/dovecot/sieve-global/after/spam_to_junk.sieve << EOF
+require ["fileinto","mailbox"];
+
+if anyof (header :contains "X-Spam-Flag" "YES",
+          header :contains "X-Spam" "Yes") {
+    fileinto "Junk";
+}
+EOF
```

Comment:
Honestly not a fan of `EOF` like this vs separate files, but I've said this before with cron jobs and looked the other way in the PR where this was also done :sweat_smile: 

Separate config files have better observability in my opinion.
- Far more visible in the file tree as distinct content from scripts. Also beneficial for tasks like `grep` on content that isn't `.sh` when searching / filtering the entire project for something, less noise to wade through.
- `git blame` far more useful as they don't tend to go through as much churn as scripts have been (_each year it seems they get refactored heavily_).

I don't think there's much gain from inlining content like this, but I know it's preferred by you and casper :sweat_smile:

Reply:
I know you're not a fan, but from my perspective, the gains (reduced complexity in the file tree, Dockerfile and removal of the uncanny `rm` in the `else` branch) outweigh the disadvantages in this case for such a small file :) 

With larger files, I agree with you!

Reply:
> I don't think there's much gain from inlining content like this, but I know it's preferred by you and casper 😅

I depends on 😆 For static content, I generally prefer plain files. For small snippets (with variables), a heredoc may better suited.
If there is consent, we can in a separate PR move some (not only rspam related) heredocs to files.

Reply:
Agreed!

Reply:
> If there is consent, we can in a separate PR move some (not only rspam related) heredocs to files.

I can live with it :sweat_smile: 

There's some merit for the co-location to that extent perhaps, but those less familiar where everything is and wanting to contribute or troubleshoot something may first expect files in the container that we add to exist as files in our project, rather than embedded into scripts :man_shrugging: (_I know this was a bit of a pain-point for me with the cron jobs, which I think existed in several locations in the container, some from packages beyond what DMS added_)

It probably won't make much of a difference to getting new maintainers onboard though, so if it's easier for those actively maintaining the project to leverage heredocs, go for it :sunglasses: